File: 01 - Multiples of 3 and 5/Python/MultiplesOf3And5.py
# Project Euler - Problem 1

def sumInRange(value, rangeMin, rangeMax):
	return sum([value * i for i in range(rangeMin / value + 1, (rangeMax - 1) / value + 1)])

def main():
	print "Sum: ", sumInRange(3,0,1000) + sumInRange(5,0,1000) - sumInRange(15,0,1000)

if __name__ == "__main__":
	main()

File: 01 - Multiples of 3 and 5/Scala/MultiplesOf3And5.scala
object MultiplesOf3And5 {
	def sumInRange(value: Int, maxRange: Int): Int = {
		return (0 to ((maxRange-1) / value)).map(_ * value).sum
	}

	def main(args: Array[String]) {
		println(sumInRange(3, 1000) + sumInRange(5, 1000) - sumInRange(15, 1000))
	}
}